### PR TITLE
TOOLS-2225 mongorestore 4.2 transaction support

### PR DESCRIPTION
**Pull request by xdg**
This CR is based after the revendoring of mongo-tools-common to bring in the transaction oplog buffer from TOOLS-2291

### CODE REVIEW

File: mongorestore/mongorestore_txn_test.go [hunk 1]
```diff
@@ -0,0 +1,171 @@
...
+		So(result.Err, ShouldBeNil)
+
+		for k, v := range data {
+			Println("postImageCheck for", k)
+			So(postImageCheck(client, v), ShouldBeNil)
+		}
+	})
+}
+
+func createTxnTestDataDir(t *testing.T, data txnTestDataMap) string {
```
Comment (kmahar) [inline mongorestore/mongorestore_txn_test.go]: just to make sure I understand: this is combining all of the oplog entries from the different test cases in the JSON file so they are applied in some interleaved order, right? (so basically all the tests are happening at once)

File: mongorestore/fixtures_test.go [hunk 2]
```diff
@@ -0,0 +1,167 @@
...
+		panic(fmt.Sprintf("invalid namespace '%s'", tcd.ns))
+	}
+	return ns[0], util.EscapeCollectionName(ns[1])
+}
+
+func (tcd testCollData) Mkdir(basePath string) error {
+	db, _ := tcd.SplitNS()
+	dbDir := path.Join(basePath, db)
+	err := os.MkdirAll(dbDir, 0755)
+	if err != nil {
```
[the remark this answers is not in the document]
Comment (xdg) [reply mongorestore/fixtures_test.go]: Good catch.  I think I was doing something else after and deleted it.

File: mongorestore/fixtures_test.go [hunk 3]
```diff
@@ -0,0 +1,167 @@
+// Copyright (C) MongoDB, Inc. 2019-present.
```
[the remark this answers is not in the document]
Comment (xdg) [reply mongorestore/fixtures_test.go]: Good point.  Done.

File: mongorestore/mongorestore_txn_test.go (hunk 1 above)
Comment (xdg) [reply mongorestore/mongorestore_txn_test.go]: Yes, added a comment to that effect.

**Pull request closed without merging**